### PR TITLE
[58_7] Add CI for archlinux

### PR DESCRIPTION
**Pull request by Oyyko**
## Why you open this Pull Request?
Add CI for xmake build on Archlinux.

### CODE REVIEW

File: .github/workflows/ci-xmake-archlinux.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - 'src/**'
+      - '!src/Plugins/Macos/**'
+      - '!src/Plugins/Windows/**'
+      - 'tests/**'
+      - 'misc/xmake/packages.lua'
+      - 'xmake.lua'
+      - '.github/workflows/ci-archlinux.yml'
+
+env:
+          XMAKE_ROOT: y
```
Comment (darcy-shen) [inline .github/workflows/ci-xmake-archlinux.yml:25]: indent less to make it beautiful?

File: .github/workflows/ci-xmake-archlinux.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        run: |
+          sudo pacman -Syyu --noconfirm --needed
+          sudo pacman -S --noconfirm --needed qt6-base qt6-svg sqlite libpng zlib libjpeg curl texlive-core python libxext xmake unzip
+      - uses: actions/checkout@v1
+      - name: Xrepo update
+        run: yes | XMAKE_ROOT=y xrepo update-repo
+      - name: Config
+        run: XMAKE_ROOT=y xmake config -m release --yes
+      - name: Build  
+        run: XMAKE_ROOT=y xmake build --yes --verbose --diagnosis --jobs=8 --all
```
Comment (darcy-shen) [inline .github/workflows/ci-xmake-archlinux.yml:48]: newline, please

File: .github/workflows/ci-xmake-archlinux.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - name: PacMan
+        env:
+          XMAKE_ROOT: y
+        run: |
+          sudo pacman -Syyu --noconfirm --needed
+          sudo pacman -S --noconfirm --needed qt6-base qt6-svg sqlite libpng zlib libjpeg curl texlive-core python libxext xmake unzip
+      - uses: actions/checkout@v1
+      - name: Xrepo update
+        run: yes | XMAKE_ROOT=y xrepo update-repo
```
Comment (darcy-shen) [inline .github/workflows/ci-xmake-archlinux.yml:44]: `xmake repo --update`
可以用这个

File: .github/workflows/ci-xmake-archlinux.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Config
+        run: XMAKE_ROOT=y xmake config -m release --yes
```
Comment (darcy-shen) [inline .github/workflows/ci-xmake-archlinux.yml:46]: 移除 XMAKE_ROOT

File: .github/workflows/ci-xmake-archlinux.yml (hunk 2 above)
Comment (darcy-shen) [inline .github/workflows/ci-xmake-archlinux.yml:48]: 移除XMAKE_ROOT, `--verbose --diagnosis`改为`-vD`

**Comment by Oyyko** [comment]
GOT

File: .github/workflows/ci-xmake-archlinux.yml (hunk 2 above)
Comment (Oyyko) [reply .github/workflows/ci-xmake-archlinux.yml:48]: newline on where?

**Comment by Oyyko** [comment]
did not trigger CI, I will fix it.

**Comment by Oyyko** [comment]
Maybe a new PR will work.

**Pull request closed without merging**